### PR TITLE
v2 Tab Bar: Implementing automatic text adjustment to prevent overflow or multiline issues and removing padding on smaller devices.

### PR DESCRIPTION
**Pull request by Anubhvv**
### Problem 
The text was overflowing and being present in multiple lines for tab items in tab bar. 
Requirement is for text to remain in one line. 

There are 2 things we can do here:
For any screen width smaller than 360dp, the side padding (16dp on either sides) will reduce proportionately to accommodate text. 

 
2. We can reduce text size in accordance to fit in one line. What we do on Title bar currently.
### Root cause 

By design

### Fix

1.  we are detecting text overflow using the onTextLayout callback. This allows us to recursively reduce text size untill it fits in the avaibale one line width
2. For small devices have removed the 16.dp padding

### Validations

(how the change was tested, including both manual and automated tests)

### Screenshots

| small phone                          | big phone                                   |
|----------------------------------------------|--------------------------------------------|
|image| image |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] Automated Tests
- [ ] Documentation and demo app examples
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 1]
```diff
@@ -202,11 +208,22 @@ fun TabItem(
                 .then(widthModifier)
         ) {
             badgeWithIcon()
+            var fontSize = remember { mutableStateOf(14.sp) }
+            var textStyle by remember(textColor) { mutableStateOf(TextStyle(color = textColor, textAlign = TextAlign.Center, fontSize = fontSize.value)) }
+
             if (textAlignment == TabTextAlignment.VERTICAL) {
                 Spacer(modifier = Modifier.height(2.dp))
-                BasicText(
+                Text(
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:216]: Lets not use material related APIs. Any reason why Basic text is not used. basictext is compose API

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 2]
```diff
@@ -202,11 +208,22 @@ fun TabItem(
                 .then(widthModifier)
         ) {
             badgeWithIcon()
+            var fontSize = remember { mutableStateOf(14.sp) }
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:211]: Why 14.sp for fontsize?

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt [hunk 3]
```diff
@@ -59,7 +60,7 @@ fun TabBar(
         )
         Row(
             modifier = Modifier
-                .padding(horizontal = 16.dp)
+                .padding(horizontal = if(LocalConfiguration.current.screenWidthDp.dp > 360.dp )16.dp else 1.dp)       // For small phones we don't want the text overflowing, hence removing padding
```
Comment (praveen970) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt:63]: What if the device is small and the number of items are 2. There will be no padding and is this expected? Should we removed the padding only when the items are more?

Comment (Anubhvv) [reply fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt:63]: small phone
There's plenty of space in less number of items

**Review by praveen970 — APPROVED** [review]

**Pull request merged**